### DISCUSSION TITLE
SvelteKit NextAuth to Auth0 stopped working

### DISCUSSION THREAD

**Discussion by sneakyPad** [vercel/community#4591] [Bug Report]
### Summary

Hi,
I've been using NextAuth with auth0 for the past few weeks without any problems. However, today I noticed that the authentication flow is no longer working. I configured Auth0 to use Google as the social login. 

I cannot replicate this locally.

Here's what I tried:
- Social login test works on Auth0
- Running my app locally works fine for both my tenants (production and dev)
- Downgrading to a previous version of next-auth and @auth/sveltekit did not help (same with upgrading)
- I went back in my deployment history and redeployed to a previous version where I was sure it would work (they all don't).
- Read about the bug mentioned in the bug log: https://errors.authjs.dev#signinerror , but I have verified that Google works as social auth.

Based on the error, I would assume that the problem is on Auth0, e.g. an accidental breaking change. However, this should also affect my locally running app, but locally it works fine.

Has anything changed on Vercel's end or any idea how I could dig deeper? 

Thanks!

### Steps to Reproduce

Here's the snippet of how I set up the Authentication handler in my `hooks.server.js`:
```js
export const handle = SvelteKitAuth({
  providers: [Auth0({ clientId: AUTH0_REG_WEBAPP_CLIENT_ID, 
        clientSecret: AUTH0_REG_WEBAPP_CLIENT_SECRET, 
        domain: AUTH0_DOMAIN,
        issuer: `https://${AUTH0_DOMAIN}`,
        authorization: { params: { scope: "openid profile email offline_access",
         prompt: "login", 
         audience: `https://${AUTH0_DOMAIN}/api/v2/`} },
      })],
```

Here's what I get after clicking on Sign In:
![iScreen Shoter - Google Chrome - 231024160610](https://github.com/vercel/community/assets/23212901/4e56020f-c997-4d49-b0e3-1d5caf84813e)

Here is the log:

```
[31m[auth][error][SignInError][0m: Read more at https://errors.authjs.dev#signinerror
[31m[auth][cause][0m: TypeError: "response" must be an instance of Response
at Module.processDiscoveryResponse (file:///var/task/node_modules/.pnpm/oauth4webapi@2.3.0/node_modules/oauth4webapi/build/index.js:212:15)
at getAuthorizationUrl (file:///var/task/node_modules/.pnpm/@auth+core@0.14.0/node_modules/@auth/core/lib/oauth/authorization-url.js:19:28)
at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
at async Module.signin (file:///var/task/node_modules/.pnpm/@auth+core@0.14.0/node_modules/@auth/core/lib/routes/signin.js:15:20)
at async AuthInternal (file:///var/task/node_modules/.pnpm/@auth+core@0.14.0/node_modules/@auth/core/lib/index.js:112:36)
at async Auth (file:///var/task/node_modules/.pnpm/@auth+core@0.14.0/node_modules/@auth/core/index.js:100:30)
at async respond (file:///var/task/.svelte-kit/output/server/index.js:3289:22)
at async Server.default (file:///var/task/.svelte-kit/vercel-tmp/index.js:47:3)
at async Server.<anonymous> (/var/task/___vc/__launcher/__launcher.js:14:11367)
[31m[auth][details][0m: {
"provider": "auth0"
}```

**Comment by sneakyPad** [comment]
Small update: I have deployed the exact same application to Netlify to rule out Auth0 being an issue and the authentication flow works fine on it. Based on this I would assume it is an issue with Vercel itself. I can build a small app and make it public if needed?

**Comment by pupvogel** [comment]
Looks like I have this problem too, using both Google and Github-providers the Vercel-ServerLogs show the same thing
```
TypeError: "response" must be an instance of Response
at Module.parseWwwAuthenticateChallenges .......
```
For Google that error shows up as "[auth][error][SignInError]" on "[POST] /auth/signin/google",
for Github it shows up as "[auth][error][CallbackRouteError]" on "[GET] /auth/callback/github".

Any help appreciated..!

**Comment by johnyvelho** [comment]
same here. I've delved into the issue and pinpointed that the error arises specifically during the execution of the `processDiscoveryResponse` method from the `oauth4webapi` package. more details here:

https://github.com/nextauthjs/next-auth/discussions/8940

**Comment by johnyvelho** [comment]
Replicating here:

> I hit a dead end.
> this is the function causing the issue from oauth4webapi:
> 
> ```javascript
> export async function processDiscoveryResponse(expectedIssuerIdentifier, response) {
>     if (!(expectedIssuerIdentifier instanceof URL)) {
>         throw new TypeError('"expectedIssuer" must be an instance of URL');
>     }
>     if (!(response instanceof Response)) {
>         throw new TypeError('"response" must be an instance of Response');
>     }
>     if (response.status !== 200) {
>         throw new OPE('"response" is not a conform Authorization Server Metadata response');
>     }
>     assertReadableResponse(response);
>     let json;
>     try {
>         json = await response.json();
>     }
>     catch {
>         throw new OPE('failed to parse "response" body as JSON');
>     }
>     if (!isJsonObject(json)) {
>         throw new OPE('"response" body must be a top level object');
>     }
>     if (!validateString(json.issuer)) {
>         throw new OPE('"response" body "issuer" property must be a non-empty string');
>     }
>     if (new URL(json.issuer).href !== expectedIssuerIdentifier.href) {
>         throw new OPE('"response" body "issuer" does not match "expectedIssuer"');
>     }
>     return json;
> }
> ```
> 
> the error is coming from this block:
> `if (!(response instanceof Response)) {
>         throw new TypeError('"response" must be an instance of Response');
>     }
> `
> 
> printing the constructor name both in local and Vercel shows 'Response'. however, in production this condition returns false for some reason. I went back several versions on `@auth/core` and `oauth4webapi` and at least this part seems the same.
> Since locally works normally, I am assuming Vercel is changing the Response somehow in order to change the instanceof Response.
> I am considering change to another host or Lucia js.
> 
> the mainterners probably would have a good idea what the issue is, so maybe they will help us.
